src/lib/pages/backoffice/DocumentRequest/DocumentRequestDetails/DocumentRequestSteps/ProviderStep/ProviderStep.js

```diff
@@ -119,25 +118,34 @@ AcqProvider.propTypes = {
 };
 
 class IllProvider extends Component {
-  onCreateClick() {
-    return goTo(ILLRoutes.borrowingRequestCreate);
-  }
-
-  onSelectResult = async data => {
-    // NOTE: do the same things as AcqProvider.onSelectResult()
+  onSelectResult = async loanData => {
+    const {
+      data: { pid },
+      fetchDocumentRequestDetails,
+    } = this.props;
+    const { ill } = invenioConfig.documentRequests.physicalItemProviders;
+    const resp = await documentRequestApi.addProvider(pid, {
```

Comment:
I would suggest to refactor this code that is done 2 times and actually to move it to redux. This is because it is doing 2 requests:
1. add provider
2. as a side effect, re-fetch the details
This should be an atomic operation done in redux.